### DISCUSSION TITLE
Multiple Search Result Panes

### DISCUSSION THREAD

**Discussion by chucklepie** [godotengine/godot-proposals#5490] [Editor]
Currently when you search in the project you get a single Search Results pane. If you then want to look for something else then come back, you can't you've lost your search. I do this all the time, not just in Godot but in my job. I imagine everyone else does.

Most other IDE's, including VSCode, Visual Studio, Intelli, etc have this functionality. 

For example I search for 'dialogue_box':
![image](https://user-images.githubusercontent.com/12863685/192140940-c44a8c10-37f9-4bb4-bdf7-4875026eed37.png)

Now, I want to search for all instances of what I found, e.g. "single-bullet" then go back to my original search and continue.  If do this, I've lost my original search so have to enter it again, remember where I was, find it in the results...

In an editor like VS Code, for example I search for the same 'dialogue_box':
![image](https://user-images.githubusercontent.com/12863685/192141040-4785b678-7466-45e3-bd44-92357fb788f4.png)

I can now just open another box and search for that item 'single_bullet' in another pane:
![image](https://user-images.githubusercontent.com/12863685/192141058-f0ace377-7922-4d1f-84e6-b1e5a62a8cd4.png)

But I still have my original ones open.

This will/could double as a search history too.

**Comment by Calinou** [comment]
The way VS Code does advanced search results panes is by creating a specially formatted text file, then opening it in the editor. There's a dedicated syntax highlighting language for highlighting those files, and <kbd>Ctrl + clicking</kbd> file names is possible as VS Code can recognize those relative paths when written in files.

It's awkward, but it allowed the VS Code team to reuse the existing editor features such as selecting text, etc.

**Reply by chucklepie** [reply]
I mentioned vscode with pictures because I was on my other computer to show the intent on and benefit only.

Regardless of how vscode works, and it is clunky compared to the beauty of visual studio, I'm just saying multiple search result tabs would be very beneficial.
